### ISSUE TITLE
Unexpected behavior around Activity external completion with CanceledError.

### ISSUE THREAD

**Issue by johncburns1** [nforgeio/neonKUBE#892]
The Cadence docs clearly state that Activities can be completed externally with either a result or any of the 3 error types: 
```
//  - ErrorWithDetails
//  - TimeoutError
//  - CanceledError
```
Completing an Activity with a `TimeoutError` or `ErrorWithDetails` succeeds and fails the running Activity with the specified error.  Completing an Activity externally with a `CanceledError` causes the Activity to lock up and timeout.  I am not sure if this is the expected behavior or not.  One would think that canceling an Activity would cause the Activity to return with a `CanceledError` so this could be relayed to the parent Workflow, but I could also see Activity cancellation being a fire-and-forget operation.  

I am going to test this in simple native Golang Workflow and see what the behavior is.

**Comment by johncburns1** [comment]
After my conversation with @jefflill it looks like this is the expected behavior and the issue is in the heartbeat reply back to the .NET client.  Further investigation is needed.

**Comment by jefflill** [comment]
**FIXED** for v2.2.0

Jack did a bunch of research on this and we talked it through.  The conclusion was that it appears that Cadence doesn't support cancelling an external activity by ID or Token and that although activities can be completed externally, they cannot be cancelled externally.  The parent workflow needs to do this.

I've removed the Cadence/Temporal client `ActivityCancelByTokenAsync()` and `ActivityCancelByIdAsync()` methods and the related unit tests.
